Assets/Scripts/PlayerController.cs

```diff
@@ -100,4 +101,8 @@ void Update()
     public Vector2 GetPosition() {
         return rb.position;
     }
+
+    public List<Vector2> GetUnitsPositions() {
+        return unitList.Select(unit => new Vector2(transform.position.x, transform.position.y)).ToList();
```

Comment:
You forgot to get the position of the units, and are instead getting the player's position.
Should be `unit.transform.position` instead of just `transform.position`